.github/workflows/lint.yml

```diff
@@ -0,0 +1,85 @@
+name: lint
+
+on:
+  push:
+    branches:
+      - main
+      - 'release/**'
+  pull_request:
+
+env:
+  GO_VERSION: 1.23.x
+
+jobs:
+  go:
+    timeout-minutes: 5
+    name: "go | ${{ matrix.goos }} | ${{ matrix.gonext }}"
+    runs-on: "${{ matrix.os }}"
+    defaults:
+      run:
+        shell: bash
+    strategy:
+      matrix:
+        include:
+          - os: ubuntu-24.04
+            goos: linux
+          - os: ubuntu-24.04
+            goos: freebsd
+          # FIXME: this is currently failing in a non-sensical way, so, running on linux instead...
+          # - os: windows-2022
+          - os: ubuntu-24.04
+            goos: windows
+          - os: ubuntu-24.04
+            goos: linux
+            # This allows the canary script to select any upcoming golang alpha/beta/RC
+            gonext: next
+          - os: ubuntu-24.04
+            goos: freebsd
```

Comment:
Probably no need to enable canary for non-Linux

Reply:
These are really fast (less than a minute and they are run in parallel).

I appreciate your point that they might not catch _many_ issues (as it would have to be specific to something for that platform, for the upcoming go version), but then, they are not doing harm.

Happy to remove them though - not a strong opinion - just lmk?

Reply:
I'd prefer to remove them, as they might still increase the flakiness depending on the network condition of the GHA infra.
It is also harder to scroll the CI jobs in the web browser.

Reply:
Sure, removing now.

> It is also harder to scroll the CI jobs in the web browser.

Yeah, we might want to remove the use of the github action for golintci and instead just use the binary so that it would ONE task. There is no need for it to appear as separate task IMHO, but that is how they structure their GH action.